### DISCUSSION TITLE
Can I limit which C# assemblies, classes, methods and properties are available to Python?

### DISCUSSION THREAD

**Discussion by asthomas** [pythonnet/pythonnet#2387] [General]
I would like to embed Python in an application using python.net.  Ideally, I would like Python to run in a limited environment (I hesitate to use the term "sandbox") where some assemblies, classes and members are off-limits.

I have embedded other languages, and some provide callbacks that I can hook whenever an assembly, class or member is being mapped into the interpreter.  The callback gives me the option of denying the mapping, so I can implement whatever logic I like to create blacklists, whitelists and pattern-based denials.

Is there anything in python.net that could be used to do something similar?

**Comment by Trigve** [comment]
I'm too interested to be able to specify which class should be "exported" or which methods, properties, ... . Is it even possible?
